### DISCUSSION TITLE
Possible to integrate IR in the app?

### DISCUSSION THREAD

**Discussion by Tiwas** [OlivierZal/com.melcloud#63] [Ideas]
For my setup, my AC should be within IR reach of Homey. In order to speed up actions, would it be possible to have an IR option in the app? Also, do you know if the Mitsubishi AC units supports discrete codes? The remote doesn't suggest they do, but - maybe someone's tried?

**Comment by OlivierZal** [comment]
Hi @Tiwas,

What do you mean by "IR reach of Homey" / IR option and discrete code? Maybe I know but not under those English terms...

**Reply by Tiwas** [reply]
My Homey is positioned so that the aircon can receive IR signals from it. So - instead of waiting for melcloud to send updates (some times fast, some times very slow) IR would be an option. Even sending the signal by IR first and then through Melcloud would speed things up.

Discreet code is usually when you can send a signal with a specific purpose.

So, one signal for on and another signal for off. Some have the same signal for both, so you need to know the state. If all codes are discreet, it would also be possible to send "heat" and "cool" as commands, instead of hoping that sending the "mode" signal would make the aircon end up in the right mode (requires that you know which mode it's already in, unless you have a discreet code/signal).

I hope that made it more clear. I'm on my way to bed, so please excuse me if I'm incoherent 🤣

**Comment by OlivierZal** [comment]
Ok, I understand better, thanks :-)

So Homey can handle radio signals and infrared: https://apps.developer.homey.app/wireless/infrared

That's not the logic of the MELCloud app, which relies on the (MEL)Cloud API, so it would be a totally different logic for which there is no documentation – as far as I know, but which is interesting since indeed it would work locally, without depending on the vagaries of Wi-Fi / network or API bugs. 

Note that this method would not work for multi-device users, unless you have one Homey per device, although the MELCloud API can handle them all.

By the way, some products do that, so technically it's possible
- https://fr.shop.tado.com/products/smart-ac-control-v3
- https://sensibo.com/
I've tried the first one, but it does support all the heat pump capabilities (e.g. only 3 fan speeds, no vane direction control...) and I would be forced to buy 8 since I have 8 units. Moreover open window detection was not reliable.

To do some automation: some Homey apps exist to connect those products to Homey (https://homey.app/fr-fr/app/com.tado2/tado°/) but still: you need to buy a product in addition to Homey Pro and it does not cover all the heat pump features.

Kind regards
